nomad/client_fs_endpoint.go

```diff
@@ -0,0 +1,449 @@
+package nomad
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"time"
+
+	metrics "github.com/armon/go-metrics"
+	"github.com/hashicorp/nomad/acl"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/ugorji/go/codec"
+)
+
+// FileSystem endpoint is used for accessing the logs and filesystem of
+// allocations from a Node.
+type FileSystem struct {
+	srv *Server
+}
+
+func (f *FileSystem) register() {
+	f.srv.streamingRpcs.Register("FileSystem.Logs", f.logs)
+	f.srv.streamingRpcs.Register("FileSystem.Stream", f.stream)
+}
+
+func (f *FileSystem) handleStreamResultError(err error, code *int64, encoder *codec.Encoder) {
+	// Nothing to do as the conn is closed
+	if err == io.EOF || strings.Contains(err.Error(), "closed") {
+		return
+	}
+
+	// Attempt to send the error
+	encoder.Encode(&cstructs.StreamErrWrapper{
+		Error: cstructs.NewRpcError(err, code),
+	})
+}
+
+// List is used to list the contents of an allocation's directory.
```

Comment:
Maybe to be extra clear, include in each method's comment that these methods are just pass-throughs to the client.

Reply:
Or at the top of file or something.